src/libraries/System.Configuration.ConfigurationManager/src/System/Configuration/ClientConfigPaths.cs

```diff
@@ -83,7 +91,10 @@ private ClientConfigPaths(string exePath, bool includeUserConfig)
 
             if (!string.IsNullOrEmpty(ApplicationUri))
             {
-                ApplicationConfigUri = ApplicationUri + ConfigExtension;
+                // on Unix, we want to first append '.dll' extension and on Windows change '.exe' to '.dll'
+                // eventually, in ApplicationConfigUri we will get '{applicationName}.dll.config'
+                string applicationPath = Path.ChangeExtension(ApplicationUri, ".dll");
```

Comment:
Is this going to break applications that have `.` in the name on Unix?

Reply:
Windows check was missing, added.